source/Messaging/source/Messaging.IntegrationTests/ServiceBusTopicSubscriptionDeadLetterHealthCheckTests.cs

```diff
@@ -0,0 +1,136 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Azure.Messaging.ServiceBus;
+using Energinet.DataHub.Core.Messaging.Communication.Extensions.Builder;
+using Energinet.DataHub.Core.Messaging.IntegrationTests.Fixtures;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Energinet.DataHub.Core.Messaging.IntegrationTests;
+
+public sealed class ServiceBusTopicSubscriptionDeadLetterHealthCheckTests(ServiceBusFixture fixture)
+    : IClassFixture<ServiceBusFixture>, IAsyncLifetime
+{
+    private const string HealthCheckName = "Some_Health_Check_Name";
+
+    private ServiceCollection Services { get; } = new();
+
+    private ServiceBusFixture Fixture { get; } = fixture;
+
+    public Task InitializeAsync()
+    {
+        // No-op
+        return Task.CompletedTask;
+    }
+
+    public async Task DisposeAsync()
+    {
+        var receivedMessage = await Fixture.Receiver!.ReceiveMessageAsync();
+        if (receivedMessage is not null)
+        {
+            await Fixture.Receiver!.CompleteMessageAsync(receivedMessage);
+        }
+
+        var checkMessageReceiver = await Fixture.Receiver!.ReceiveMessageAsync(TimeSpan.FromSeconds(1));
+        if (checkMessageReceiver != null)
+        {
+            throw new InvalidOperationException("Message was not removed from the topic.");
+        }
+
+        var deadLetterMessage = await Fixture.DeadLetterReceiver!.ReceiveMessageAsync();
+        if (deadLetterMessage != null)
+        {
+            await Fixture.DeadLetterReceiver!.CompleteMessageAsync(deadLetterMessage);
+        }
+
+        var checkMessageDeadLetter = await Fixture.DeadLetterReceiver!.ReceiveMessageAsync(TimeSpan.FromSeconds(1));
+        if (checkMessageDeadLetter != null)
+        {
+            throw new InvalidOperationException("Message was not removed from the dead letter queue.");
+        }
+    }
+
+    [Fact]
+    public async Task Given_MessageAddedToServiceBusTopic_Then_Healthy()
+    {
+        // Arrange
+        var healthChecksBuilder = Services
+            .AddLogging()
+            .AddHealthChecks();
+
+        healthChecksBuilder.AddServiceBusTopicSubscriptionDeadLetter(
+            _ => Fixture.ServiceBusResourceProvider.FullyQualifiedNamespace,
+            _ => Fixture.TopicResource!.Name,
+            _ => Fixture.TopicResource!.Subscriptions.First().SubscriptionName,
+            _ => Fixture.AzureCredential,
+            HealthCheckName);
+
+        // Act
+        var sender = Fixture.TopicResource!.SenderClient;
+        var message = new ServiceBusMessage("Test message");
+        await sender.SendMessageAsync(message);
+
+        // Assert
+        var provider = Services.BuildServiceProvider();
+        var healthReport = await provider
+            .GetRequiredService<HealthCheckService>()
+            .CheckHealthAsync();
+
```

Comment:
I believe this is the "act"
```suggestion
        var actualHealthReport = await provider
            .GetRequiredService<HealthCheckService>()
            .CheckHealthAsync();

```